rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+- `r` a container of resources of type `std::vector<resource_t<T>`.
+
+| *Must* be well-formed | Description |
+| --------------------- | ----------- |
+| `b.submit(s, f, args…)` | Returns an object that satisfies *Submission*. The function invokes `f` but does not wait for the `wait_t<T>` object returned by it. |
+| `b.get_submission_group()` | Returns an object that has a member function `void wait()`. Calling this wait function blocks until all previous submissions to this backend are complete. |
+| `b.get_resources()` | Returns a `std::vector<resource_t<T>>`. |
+
+| *Optional* | Description |
+| --------------------- | ----------- |
+| `void lazy_report()` | If defined by a backend, this function must be called before each new selection. It triggers reporting of the necessary execution info back to the policy. |
+
+<a id="free_functions_id"></a>
+## Free Functions
+
+| Signature | Description |
+| --------- | ----------- |
+| `vector<typename policy_traits<P>::resource_type> get_resources(P&& p);` | Returns the resources associated with the Policy `p`. |
+| `template<typename P, typename... Args> selection_t<P> select(P&& p, Args&&... args);` | Applies the policy `p` and returns a *Selection*. |
+| `template<Selection S, tyepname F, typename... Args> auto submit(Selection s, F&& f, Args&&... args);` | Invokes `f` with the unwrapped resource from selection `s` and `args`. Implements any instrumentation necessary for the backend to report necessary execution information. May be implemented as `s.get_policy().submit(s, f, args…)`. |
+| `template<Policy P, tyepname F, typename... Args> auto submit(P&& p, F&& f, Args&&... args);` | Invokes `f` with the unwrapped resource returned by `select(p, f, args…)` and `args`. Implements any instrumentation necessary for the backend to report necessary execution information. May be implemented as `p.submit(p.select(p, f, args…), f, args…)`. |
+| `template<Selection S, tyepname F, typename... Args> auto submit_and_wait(Selection s, F&& f, Args&&... args);` | Invokes `f` with the unwrapped resource from selection `s` and `args`. And then waits on object returned by the `f`.  May be implemented as `wait(s.get_policy().submit(s, f, args…))`. |
+| `template<Policy P, tyepname F, typename... Args> auto submit_and_wait(P&& p, F&& f, Args&&... args);` |  Invokes `f` with the unwrapped resource returned by `select(p, f, args…)` and `args`.And then waits on object returned by the `f`. May be implemented as `wait(p.submit(p.select(f, args…),f,args…))`. |
+| `template<typename P> auto get_submission_group(P&& p);` | Returns an object that has a member function `void wait()`. Calling this wait function blocks until all previous submissions to this policy are complete. |
+| `template<typename W> void unwrap(W&& w) noexcept;` | Returns `w.unwrap()` if available, otherwise returns `w`. |
+| `template<typename W> void wait(W&& w);` | Calls `w.wait()` if available. |
```

Comment:
As implemented, I believe this has a build error if `w.wait()` is unavailable, so the language here is a bit confusing.  
I think we need to remove "if available" here, or implement some alternative if it is not available (no-op?).

Reply:
Will remove "if available"